.github/actions/find-intermediates-base/README.md

```diff
@@ -0,0 +1,40 @@
+# Find Intermediates Base Action
+
+This GitHub Action finds the most recent commit (from the last 5) with cached build artifacts for a specified Nix flake `packages` output.
+
+## Dependencies
+
+- **Nix**: Required for checking if the binary cache has intermediates cached.
+- **Git**: Required for `git log` to retrieve commit history.
+- **Bash**: The Action uses Bash scripting, so a Unix-like environment is required.
+
+
+### Example Workflow
+
+```yaml
+name: CI with Cached Intermediates
+on:
+  pull_request:
+jobs:
+  find-intermediates:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install Nix
+        uses: DeterminateSystems/nix-installer-action@v1.0.0
+      - name: Find Build Intermediates Base
+        if: github.event_name == 'pull_request'
+        id: find_build_intermediates
+        uses: ./.github/actions/find-intermediates-base
+        with:
+          cache_url: "https://cache.nixos.org"
```

Comment:
Can you explicitly specify `package_name`?

Reply:
And use a different cache, `https://cache.mycompany.org`

Reply:
https://github.com/juspay/haskell-incremental-build-template/pull/5/commits/b5d95bbd51acb4b8c320fd1414b1d06c941e932a

Reply:
> And use a different cache, https://cache.mycompany.org

https://github.com/juspay/haskell-incremental-build-template/pull/5/commits/e6ab0971b14e21d14a4fc6d0eb1490f46a78aa5c